packages/angular/ssr/src/app.ts

```diff
@@ -70,18 +79,30 @@ export class AngularServerApp {
    *
    * @param request - The incoming HTTP request to be rendered.
    * @param requestContext - Optional additional context for rendering, such as request metadata.
-   * @param serverContext - The rendering context.
    *
    * @returns A promise that resolves to the HTTP response object resulting from the rendering, or null if no match is found.
    */
-  render(
-    request: Request,
-    requestContext?: unknown,
-    serverContext: ServerRenderContext = ServerRenderContext.SSR,
-  ): Promise<Response | null> {
+  render(request: Request, requestContext?: unknown): Promise<Response | null> {
     return Promise.race([
       this.createAbortPromise(request),
-      this.handleRendering(request, requestContext, serverContext),
+      this.handleRendering(request, /** isSsrMode */ true, requestContext),
+    ]);
+  }
+  /**
+   * Renders a page based on the provided URL via server-side rendering and returns the corresponding HTTP response.
+   * The rendering process can be interrupted by an abort signal, where the first resolved promise (either from the abort
+   * or the render process) will dictate the outcome.
+   *
+   * @param url - The full URL to be processed and rendered by the server.
+   * @param signal - (Optional) An `AbortSignal` object that allows for the cancellation of the rendering process.
+   * @returns A promise that resolves to the generated HTTP response object, or `null` if no matching route is found.
+   */
+  renderStatic(url: URL, signal?: AbortSignal): Promise<Response | null> {
```

Comment:
This is a private API, and I chose not to use `prerender` for several reasons:
- The term `prerender` can be misinterpreted as something that should be executed before `render`.
- `renderStatic` clearly indicates that it renders a static version of the page without being tied to SSG usage. For example, it can also be utilized to generate app shells.

IE: `renderStatic` can be called to generate SSG and app-shell pages.

Reply:
Sounds reasonable, just as long as it's an intentional choice.